### PR TITLE
Close missing incoming branch with advice

### LINKED ISSUE #1018: Bot should post a useful comment once it detects a PR with missing repo/branch
I expect that i'd post somethig like https://github.com/ansible/ansible/pull/40411#issuecomment-418708959 if check like #1017 met.

### PR DESCRIPTION
**Pull request by webknjaz**
(not tested yet)

Fixes #1018

### CODE REVIEW

File: templates/incoming_ref_missing.j2 [hunk 1]
```diff
@@ -0,0 +1,36 @@
+Hello,
+we're closing this pull request because the original PR-bound branch
+went missing. (This PR's fork repository or branch has been deleted by
+the owner).
+
+@{{ submitter }}, if you with to restore the PR, please, re-create
+a branch in the same fork repo and with the exactly same name and reopen
+this pull request. Or submit a new one.
```
Comment (gundalow) [inline templates/incoming_ref_missing.j2]: Clarify what "a new one" means. 
A new branch and therefore PR.

File: templates/incoming_ref_missing.j2 [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+##### To whom it may concern,
+If you're an individual wishing to pick up the work done by the original
+submitter (@{{ submitter }}), [here's how you can do this](
+https://help.github.com/articles/checking-out-pull-requests-locally/#modifying-an-inactive-pull-request-locally)
+(given you followed the instructions at [Ansible Dev Guide](
+https://docs.ansible.com/ansible/devel/dev_guide/developing_rebasing.html#configuring-your-remotes)
+and have `origin` pointing to your fork and `upstream` pointing to
+`ansible/ansible`):
+
+1. :ambulance: Pull the branch locally from read-only copy held by GitHub:
```
Comment (gundalow) [inline templates/incoming_ref_missing.j2]: Please remove the emojis, in this cause I don't think they add much https://gist.github.com/gundalow/7d080d8787bc19bb11023713ebd241ec

File: templates/incoming_ref_missing.j2 (hunk 1 above)
Comment (webknjaz) [reply templates/incoming_ref_missing.j2]: A new PR. feel free to submit a "suggestion" by pressing Ctrl+G in comment box :)

File: templates/incoming_ref_missing.j2 (hunk 2 above)
Comment (webknjaz) [reply templates/incoming_ref_missing.j2]: I think they add bits of humanity to this message. I'd like to keep them. Maybe just less of them?

File: templates/incoming_ref_missing.j2 [hunk 3]
```diff
@@ -0,0 +1,36 @@
+Hello,
+we're closing this pull request because the original PR-bound branch
+went missing. (This PR's fork repository or branch has been deleted by
+the owner).
+
+@{{ submitter }}, if you with to restore the PR, please, re-create
```
Comment (webknjaz) [inline templates/incoming_ref_missing.j2]:
```suggestion
@{{ submitter }}, if you wish to restore the PR, please, re-create
```

**Comment by webknjaz** [comment]
Demo:
```console
2018-12-13 00:30:22,068 INFO url: https://github.com/ansible/ansible/pull/40411
2018-12-13 00:30:22,069 INFO title: Add support for GitHub Enterprise for github_deploy_key
2018-12-13 00:30:22,069 INFO component[f]: lib/ansible/modules/source_control/github_deploy_key.py
2018-12-13 00:30:22,069 INFO needs_revision
2018-12-13 00:30:22,070 INFO needs_revision_msg: ci failure
2018-12-13 00:30:22,070 INFO needs_rebase
2018-12-13 00:30:22,070 INFO needs_rebase_msg: merge commit 277d0011710195ca2aa715eb196deba11646d578
2018-12-13 00:30:22,071 INFO needs_rebase_msg: missing shippable.yml
{'assign': [],
 'cancel_ci': True,
 'cancel_ci_branch': True,
 'close': True,
 'close_migrated': False,
 'comments': ['Hello,\n'
              "we're closing this pull request because the original PR-bound "
              'branch\n'
              "went missing. (This PR's fork repository or branch has been "
              'deleted by\n'
              'the owner).\n'
              '\n'
              '@dcode, if you wish to restore the PR, please, re-create\n'
              'a branch in the same fork repo and with the exactly same name '
              'and reopen\n'
              'this pull request. Or submit a new PR.\n'
              '\n'
              '##### To whom it may concern,\n'
              "If you're an individual wishing to pick up the work done by the "
              'original\n'
              "submitter (@dcode), [here's how you can do this](\n"
              'https://help.github.com/articles/checking-out-pull-requests-locally/#modifying-an-inactive-pull-request-locally)\n'
              '(given you followed the instructions at [Ansible Dev Guide](\n'
              'https://docs.ansible.com/ansible/devel/dev_guide/developing_rebasing.html#configuring-your-remotes)\n'
              'and have `origin` pointing to your fork and `upstream` pointing '
              'to\n'
              '`ansible/ansible`):\n'
              '\n'
              '1. :ambulance: Pull the branch locally from read-only copy held '
              'by GitHub:\n'
              '   ```console\n'
              '   git fetch upstream '
              'pull/40411/head:recovered-feature/40411-add-support-for-github-enterprise-for-github_deploy_key\n'
              '   ```\n'
              '   Now you have it locally!\n'
              '2. :construction: Switch to your local branch containing PR '
              'changes:\n'
              '   ```console\n'
              '   git checkout '
              'recovered-feature/40411-add-support-for-github-enterprise-for-github_deploy_key\n'
              '   ```\n'
              '3. :sparkles: Do your magic. :pencil: Change stuff. :lock: '
              'Commit. :hammer:\n'
              '4. :package: Push it to your fork:\n'
              '   ```console\n'
              '   git push -u origin '
              'recovered-feature/40411-add-support-for-github-enterprise-for-github_deploy_key:recovered-feature/40411-add-support-for-github-enterprise-for-github_deploy_key\n'
              '   ```\n'
              '5. :rocket: Now, [create a PR with your renovated work](\n'
              '   '
              'https://github.com/ansible/ansible/compare/devel...:recovered-feature/40411-add-support-for-github-enterprise-for-github_deploy_key?quick_pull=1&labels=feature&title=Add '
              'support for GitHub Enterprise for github_deploy_key&body=This '
              'is a resubmission of PR %2340411).\n'
              "   _(Of course, don't forget to fill in the PR template "
              'properly)._\n'
              '\n'
              '\n'
              '<!--- boilerplate: incoming_ref_missing --->'],
 'merge': False,
 'newlabel': [],
 'open': False,
 'rebuild': False,
 'unassign': [],
 'unlabel': []}
Take recommended actions (y/N/a/R/T/DEBUG)? 
```

**Comment by felixfontein** [comment]
@webknjaz this PR seems to be broken, ansibot has been posting this message 11 times to ansible/ansible#45127 without closing it. I think/hope I stopped it with `bot_broken`, otherwise it would probably add a new message every run...

**Comment by webknjaz** [comment]
@felixfontein it's been disabled for now. I'll take a look at the fix a bit later.

**Pull request merged**